black.py

```diff
@@ -2872,7 +2872,15 @@ def normalize_invisible_parens(node: Node, parens_after: Set[str]) -> None:
             if is_walrus_assignment(child):
                 continue
             if child.type == syms.atom:
-                if maybe_make_parens_invisible_in_atom(child, parent=node):
+                # Determines if the underlying atom should be surrounded with
+                # invisible params - also makes parens invisible recursively
+                # within the atom
+                should_surround_with_parens = maybe_make_parens_invisible_in_atom(
+                    child, parent=node
+                )
+                # Removes repeated invisible parens within in the atom
+                dedupe_invisible_atom_parens(child)
```

Comment:
Is there a way we could do maybe_make_parens_invisible_in_atom() and dedupe_invisible_atom_parens() in one go, so we don't have to recurse over the expression twice?

Reply:
Yea I can try and tweak this a bit.  Thanks for feedback on the approach.